lxd/container_lxc.go

```diff
@@ -2145,12 +2145,14 @@ func (c *containerLXC) Shutdown(timeout time.Duration) error {
 
 	err = op.Wait()
 	if err != nil && c.IsRunning() {
+		op.Done(err)
 		shared.LogError("Failed shutting down container", ctxMap)
 		return err
 	}
 
 	shared.LogInfo("Shut down container", ctxMap)
 
+	op.Done(nil)
```

Comment:
Again, op.Wait() above will only return when the operation is Done, so this is also wrong.